test/Microsoft.AspNetCore.Mvc.Core.Test/Infrastructure/ConfigureCompatibilityOptionsTest.cs

```diff
@@ -0,0 +1,170 @@
+            var configure = Create(CompatibilityVersion.Version_2_0, new Dictionary<string, object>()
+            {
+                { nameof(TestOptions.TestProperty), true },
+            });
+
+            var options = new TestOptions()
+            {
+                TestProperty = false,
+            };
+
+            // Act
+            configure.PostConfigure(Options.DefaultName, options);
+
+            // Assert
+            Assert.False(options.TestProperty);
+        }
+
+        [Fact]
+        public void PostConfigure_ValueNotSet_SetsValue()
+        {
+            // Arrange
+            var configure = Create(CompatibilityVersion.Version_2_0, new Dictionary<string, object>()
+            {
+                { nameof(TestOptions.TestProperty), true },
+            });
+
+            var options = new TestOptions();
+
+            // Act
+            configure.PostConfigure(Options.DefaultName, options);
+
+            // Assert
+            Assert.True(options.TestProperty);
+        }
+
+        private static ConfigureCompatibilityOptions<TestOptions> Create(
+            CompatibilityVersion version, 
+            IReadOnlyDictionary<string, object> defaultValues)
+        {
+            var compatibilityOptions = Options.Create(new MvcCompatibilityOptions() { CompatibilityVersion = version });
+            return new TestConfigure(NullLoggerFactory.Instance, compatibilityOptions, defaultValues);
+        }
+
+        private class TestOptions : IEnumerable<ICompatibilitySwitch>
+        {
+            private readonly CompatibilitySwitch<bool> _testProperty;
+
+            private readonly ICompatibilitySwitch[] _switches;
+
+            public TestOptions()
+            {
+                _testProperty = new CompatibilitySwitch<bool>(nameof(TestProperty));
+                _switches = new ICompatibilitySwitch[] { _testProperty };
+            }
+
+            public bool TestProperty
+            {
+                get => _testProperty.Value;
+                set => _testProperty.Value = value;
+            }
+
+            public IEnumerator<ICompatibilitySwitch> GetEnumerator()
+            {
+                return ((IEnumerable<ICompatibilitySwitch>)_switches).GetEnumerator();
+            }
+
+            IEnumerator IEnumerable.GetEnumerator() => _switches.GetEnumerator();
+        }
+
+        private class TestConfigure : ConfigureCompatibilityOptions<TestOptions>
+        {
+            public TestConfigure(
+                ILoggerFactory loggerFactory,
+                IOptions<MvcCompatibilityOptions> compatibilityOptions,
+                IReadOnlyDictionary<string, object> defaultValues) 
+                : base(loggerFactory, compatibilityOptions)
+            {
+                DefaultValues = defaultValues;
+            }
+
+            protected override IReadOnlyDictionary<string, object> DefaultValues { get; }
+        }
+
+        [Fact]
+        public void Constructor_WithName_IsValueSetIsFalse()
```

Comment:
Aren't these already covered in `CompatSwitchTest.cs`?

Reply:
Oh yeah, bad copy-paste 👍